fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerIndexFromIndexTest.java

```diff
@@ -418,8 +419,315 @@ public void testIndexFromIndexPersistentPreventBadContinuation() {
+
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setIndex(tgtIndex).setSubspace(subspace)
+                .setIndexStatePrecondition(OnlineIndexer.IndexStatePrecondition.BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY_ERROR_IF_POLICY_CHANGED)
+                .setTimer(timer)
+                .build()) {
+
+            // now try building by records, a failure is expected
+            RecordCoreException e = assertThrows(RecordCoreException.class, indexBuilder::buildIndex);
+            assertTrue(e.getMessage().contains("This index was partly built by another method"));
+        }
+
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setIndex(tgtIndex).setSubspace(subspace)
+                .setTimer(timer)
+                .build()) { // IndexStatePrecondition gets the default BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY)
+
+            // now continue building, overriding the requested method with the previous one
+            indexBuilder.buildIndex(true);
+        }
+
+        // counters should demonstrate a continuation to completion
+        assertEquals(numRecords, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RECORDS_SCANNED));
+        assertEquals(numRecords, timer.getCount(FDBStoreTimer.Counts.ONLINE_INDEX_BUILDER_RECORDS_INDEXED));
+    }
+
+    @Test
+    public void testIndexFromIndexPersistentContinuePreviousByRecords() {
+        // start indexing by records, allow continuation of previous by records method - overriding the by-index request
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        final int numRecords = 1328;
+        final int chunkSize  = 42;
+
+        Index srcIndex = new Index("src_index", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS);
+        Index tgtIndex = new Index("tgt_index", field("num_value_3_indexed"), IndexTypes.VALUE);
+
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = myHook(srcIndex, tgtIndex);
+
+        openSimpleMetaData();
+        populateData(numRecords);
+
+        openSimpleMetaData(hook);
+        buildSrcIndex(srcIndex);
+
+        openSimpleMetaData(hook);
+        final AtomicLong counter = new AtomicLong(0);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setIndex(tgtIndex).setSubspace(subspace)
+                .setLimit(chunkSize)
+                .setTimer(timer)
+                .setConfigLoader(old -> {
+                    if (counter.incrementAndGet() > 7) {
+                        throw new RecordCoreException("Intentionally crash during test");
+                    }
+                    return old;
+                })
+                .build()) {
+            // build by index
```

Comment:
```suggestion
            // build by records
```

Reply:
This copy-paste trap was eliminated with the new `buildIndexAndCrashHalfway` function.